src/main/kotlin/org/jetbrains/research/ml/coding/assistant/runner/HintGenerateRunner.kt

```diff
@@ -0,0 +1,126 @@
+
+    class TransformationsRunnerArgs(parser: ArgParser) {
+        val taskName by parser.storing(
+            "-t",
+            "--task_name",
+            help = "Name of the task"
+        )
+
+        val solutionSpacePath by parser.storing(
+            "-s",
+            "--space_path",
+            help = "Path to serialized solution space"
+        )
+
+        val codeRepositoryPath by parser.storing(
+            "-c",
+            "--code_repository_path",
+            help = "Path to serialized"
+        )
+
+        val outputDir by parser.storing(
+            "-o",
+            "--output_path",
+            help = "Output directory"
+        )
+    }
+
+    @ExperimentalTime
+    override fun main(args: MutableList<String>) {
+        try {
+            ArgParser(args.drop(1).toTypedArray()).parseInto(::TransformationsRunnerArgs).run {
+                this@HintGenerateRunner.solutionSpacePath = Paths.get(solutionSpacePath)
+                this@HintGenerateRunner.codeRepositoryPath = Paths.get(codeRepositoryPath)
+                this@HintGenerateRunner.outputDir = Paths.get(outputDir)
+                this@HintGenerateRunner.datasetTask = DatasetTask.createFromString(this.taskName)
+            }
+
+            val project = ProjectUtil.openOrImport(getTmpProjectDir(), null, true)
+            project?.let { p ->
+                setSdkToProject(p, getTmpProjectDir(toCreateFolder = false))
+                val solutionSpaceFile = solutionSpacePath.toFile()
+
+                val codeRepository = createCodeRepository(codeRepositoryPath.toFile())
+                val solutionSpace = SerializationUtils.decodeSolutionSpace(solutionSpaceFile.readText())
+
+                val finder = ParallelVertexFinder(EditPartialSolutionMatcher)
+                val code = """
+print(max(input()))
+""".trimIndent()
+                val psiCreator = p.service<PsiCreator>()
+                val file = psiCreator.initFileToPsi(code)
+                val partialSolution = PartialSolution(
+                    Util.getTreeContext(file),
+                    MetaInfo(123.0f, null, 0.32, datasetTask)
+                )
+                val closestVertex = finder.findCorrespondingVertex(solutionSpace, partialSolution)
+
+                val reportGenerator = MarkdownHintReportGenerator(codeRepository)
+                val reportFile =
+                    outputDir.toFile().resolve("${datasetTask.taskName}_report.md").apply { createNewFile() }
+
+                val nextVertex = solutionSpace.graph
+                    .outgoingEdgesOf(closestVertex)
+                    .firstOrNull()
+                    ?.let(solutionSpace.graph::getEdgeTarget)
+                reportFile.outputStream().use { stream ->
+                    reportGenerator.generate(
+                        stream,
+                        HintReport(
+                            datasetTask,
+                            "ParallelVertexFinder",
```

Comment:
Maybe should you use ParallelVertexFinder::class or something like this instead a string const?